rustls/src/client/builder.rs

```diff
@@ -18,6 +18,19 @@ use core::marker::PhantomData;
 
 impl ConfigBuilder<ClientConfig, WantsVerifier> {
     /// Choose how to verify server certificates.
+    ///
+    /// Using this function does not configure revocation.  If you wish to
+    /// configure revocation, instead use:
+    ///
+    /// ```diff
+    /// - .with_root_certificates(root_store)
+    /// + .dangerous()
+    /// + .with_custom_certificate_verifier(
```

Comment:
Not a problem for this diff, but I wonder if we should add a `with_certificate_verifier_builder` option to `ConfigBuilder<ClientConfig, WantsVerifier>` that takes `ServerCertVerifierBuilder` and avoids needing `dangerous`. 

My thinking is that it's a bit unfortunate to have to go through `dangerous` if you're using the `WebPkiServerVerifier` - we've arranged its APIs to be safe. It's only providing a from-scratch `dyn ServerCertVerifier` that's dangerous.

It's a little bit unusual/awkward to accept a builder instead of a finalized struct, but I think we want to leave `ServerCertVerifierBuilder` producing `Arc<dyn ServerCertVerifier>` when it builds, so we can't take a concrete `ServerCertVerifierBuilder` type as an arg.

WDYT?

Reply:
>  I think we want to leave ServerCertVerifierBuilder producing Arc<dyn ServerCertVerifier> when it builds

Maybe that's not true. I'll fiddle a bit.

Reply:
I started on a commit today to rip out the with_roots() methods from the config builders in favor of a with_verifier(). That seems conceptually nicer (more composable) but it's also a bit of an ergonomics hit for the common case.

Reply:
Here's what I was thinking: https://github.com/rustls/rustls/pull/1609

I left the `with_root_certificates` option but implemented it in terms of `with_webpki_verifeir`.

Reply:
I am just about to hit merge on this and then go on holiday -- do you want to update this new comment in #1609?

Reply:
Sure, sounds good. We can resolve this thread.